### ISSUE TITLE
no-irregular-whitespace fails with special characters in template literals

### ISSUE THREAD

**Issue by apexskier** [eslint/eslint#6970]
**What version of ESLint are you using?**
v3.3.1

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**
```js
module.exports = {
    parser: 'babel-eslint',
    rules: {
        'no-irregular-whitespace': 2,
    },
};
```

**What did you do? Please include the actual source code causing the issue.**
```js
export default function test(a) {
    return ` •${a}`;
}

export function test2(a) {
    return ` .${a}`;
}

export function test3(a) {
    return ` \`${a}`;
}
```

**What did you expect to happen?**
This should be completely valid.

**What actually happened? Please include the actual, raw output from ESLint.**
```
.../temp/temp.js
   2:13  error  Irregular whitespace not allowed  no-irregular-whitespace
   6:13  error  Irregular whitespace not allowed  no-irregular-whitespace
  10:13  error  Irregular whitespace not allowed  no-irregular-whitespace

✖ 3 problems (3 errors, 0 warnings)
```

**Comment by apexskier** [comment]
Not actually an issue
